drivers/auxdisplay/auxdisplay_noritake_itron.c

```diff
@@ -0,0 +1,378 @@
+/*
+ * Copyright (c) 2022-2023 Jamie McCrae
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT noritake_itron
+
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/auxdisplay.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/logging/log.h>
+#include "auxdisplay_noritake_itron.h"
+
+LOG_MODULE_REGISTER(auxdisplay_noritake_itron, CONFIG_AUXDISPLAY_LOG_LEVEL);
+
+/* Display commands */
+#define NORITAKE_ITRON_CMD_USER_SETTING 0x1f
+#define NORITAKE_ITRON_CMD_ESCAPE 0x1b
+#define NORITAKE_ITRON_CMD_BRIGHTNESS 0x58
+#define NORITAKE_ITRON_CMD_DISPLAY_CLEAR 0x0c
+#define NORITAKE_ITRON_CMD_CURSOR 0x43
+#define NORITAKE_ITRON_CMD_CURSOR_SET 0x24
+#define NORITAKE_ITRON_CMD_ACTION 0x28
+#define NORITAKE_ITRON_CMD_N 0x61
+#define NORITAKE_ITRON_CMD_SCREEN_SAVER 0x40
+
+/* Time values when multithreading is disabled */
+#define NORITAKE_ITRON_RESET_TIME K_MSEC(2)
+#define NORITAKE_ITRON_RESET_WAIT_TIME K_MSEC(101)
+#define NORITAKE_ITRON_BUSY_DELAY_TIME_CHECK K_MSEC(4)
+#define NORITAKE_ITRON_BUSY_WAIT_LOOPS 125
+
+/* Time values when multithreading is enabled */
+#define NORITAKE_ITRON_BUSY_MAX_TIME K_MSEC(500)
```

Comment:
This is a very long time to de overhead to wait to learn that a display is dead. I would suggest waiting for a shorter maximum period of time. Not going to block on it though, since (presumably) the mode time to wait for the semaphore when the display is working is much, much smaller.